### PR TITLE
Add constraints for psr/cache 2.x & 3.x

### PR DESCRIPTION
**Pull request by GwendolenLynch**
<h2>Change Log</h2> 
<div id="change-log">
  <h4>Added</h4>
  <ul id="added">
    
  </ul> 
  <h4>Fixed</h4>  
  <ul id="fixed">
    
  </ul>
  <h4>Changed</h4>
  <ul id="changed">
    
  </ul>  
  <h4>Updated</h4>
  <ul id="updated">
    <li>Constraints for <code>psr-cache</code> to include v2 & v3</li>
  </ul>    
  <h4>Removed</h4>
  <ul id="removed">
    
  </ul>
  <h4>Deprecated</h4>
  <ul id="deprecated">
    
  </ul>  
  <h4>Security</h4> 
  <ul id="security">
    
  </ul>     
</div>
<hr/>

<h2>Description</h2>
<p>Add support for <code>psr/cache</code> v2 & v3 as it is required by later versions of <code>symfony/cache</code>, e.g. 6.x</p>

<p>Related to Symfony 6 support PR https://github.com/aeon-php/symfony-bundle/pull/355</p>

### CODE REVIEW

File: tools/composer.json [hunk 1]
```diff
@@ -7,5 +7,10 @@
         "phpstan/phpstan": "^1.0.0",
         "infection/infection": "^0.26.1",
         "friendsofphp/php-cs-fixer": "^3.1"
+    },
+    "config": {
+        "allow-plugins": {
+            "infection/extension-installer": true
```
Comment (norberttech) [inline tools/composer.json]: Do we need this to be set to true?

**Review by norberttech — COMMENTED** [review]
Thanks for contribution, I only have one question about allowing composer plugins, I'm not sure if its really needed.

File: tools/composer.json (hunk 1 above)
Comment (GwendolenLynch) [reply tools/composer.json]: I'm unsure. My thinking, potentially naïvely, was that this would be one of those things that caused CI to fall over.

Comment (GwendolenLynch) [reply tools/composer.json]: Yep, it was my naïvety. :innocent: 

Running `composer install --no-interaction` doesn't stop to request permission, so I have dropped that change from the commit!

**Comment by norberttech** [comment]
I'm merging this one since failing tests are not related to your changes. Thank you!

**Pull request merged**